### DISCUSSION TITLE
Broken links from markdown file in docs

### DISCUSSION THREAD

**Discussion by boomerbgit** [facebook/docusaurus#7734] [Q&A] [answered]
My issue seems to be the same as [Discussion #7246](https://github.com/facebook/docusaurus/discussions/7246).
The baseUrl is '/documentation/'
I'm running latest v2.0.0-beta.21

I have a file
/docs/application.mdx
that is trying to link to an HTML file that I copied to docs/assets/helm.html

I have tried several solutions but they all result in a broken link error-- with the same error message saying "resolved as: /documentation/docs/assets/helm.html".
`
[Deploy](/assets/helm.html)
[Deploy](./assets/helm.html)
[Deploy](assets/helm.html)
`

I also found [Issue #7247](https://github.com/facebook/docusaurus/issues/7247) and 'slorber' provided an explanation of why that broken link message is worded as it is -- resolving to a file path which exists -- but I didn't understand his explanation.

      "I understand how it can be a bit confusing here, but the "markdown file resolution" and the "broken links checked resolved as" are 2 different things somehow:
      One resolves relative file path to an absolute url
      The other resolves relative URLs to absolute URLs so that we can compare them to known site routes
      "

This is another variation of trying to solve one main issue: linking to HTML files that are not named index.html.

I have several "external" HTML files that I need to link to.  If I copy them to the 'static' folder I can link to all of the index.html files using the base folder where the index.html file is located
for example sdk/index.html can be linked to using `<a href="/sdk/">` but not using `<a href="/sdk/index.html">`
So I haven't found a way to link to a file that is not named index.html.  `<a href="/assets/helm.html">` for example.

**Comment by slorber** [comment, marked as answer]
Hey

Please provide a repro using a code playground to make it easier to understand your problem: a text makes things harder for me

---

Note your external HTML pages are not Docusaurus pages and are not rendered with React as part of the single-page-application that Docusaurus builds.

You should keep those HTML pages in `/static` for sure, otherwise, Docusaurus won't do anything with those when found in the `/docs` folder and your link will lead to nowhere.

We recommend the `pathname://` protocol to link to external HTML pages, see also: https://docusaurus.io/docs/advanced/routing#escaping-from-spa-redirects

--- 

Technically, we could consider that a link such as `[TEST](./mytest.html)` would be resolved as a regular [asset file](https://docusaurus.io/docs/markdown-features/assets#files), but this means that the URL of your HTML file will become "unpredictable" and be in practice hosted alongside other asset files (handled like docs, ppt, mp3, avi...). Those asset names are hashed to encourage aggressive/immutable HTTP caching.

For example, your URL could become something like: https://docusaurus.io/assets/files/helm-93ea653de6d8636f7254aecab1adbcce.html

---

If you want control over your HTML doc URL, I strongly  recommend keeping your HTML files in `/static` and using `pathname://`

**Reply by boomerbgit** [reply]
Thank you for the reply.
Glad to hear that external HTML pages should be in the /static folder -- that is where I started.  
I am unable to access a static file that has a name other than index.html -- and index.html is only accessed by NOT including the filename only the folder - which makes sense for default server resolution of the URL.

I had previously tried the pathname:// solution and tried it again just now -- but it still DOES NOT resolve.
I'm copying the file /cots/helm.html under docusaurus into the /static folder (static/cots/helm.html)
I tried using ```[Deployment](pathname:///cots/helm.html)```

the [escaping-from-spa-redirects page you referenced](https://docusaurus.io/docs/advanced/routing#escaping-from-spa-redirects) example shows 'pure-html' with no file extension, but then there is a subsequent example using ```[An asset from the static](pathname:///files/asset.pdf)``` so I think my syntax is correct.

**Reply by slorber** [reply]
Please prove that it doesn't work with a sandbox, you probably made a mistake and I can't tell if you don't show me any runnable env to see your problem in action

Important note: 
- the `/static` folder is copied as-is to the `/build` output on `docusaurus build`
- using `pathname://` will just render your path as a regular `<a>` link (non-SPA): Docusaurus does not do anything with it and you have the responsibility that your link works using regular html/hosting

There may be some slight differences betwees prod/dev modes (particularly related to how files are hosted, file extensions and trailing slashes) so be sure to try both: it is possible that your setup works in prod but not in dev.

**Reply by boomerbgit** [reply]
Thanks for the follow up.  I did create a sample project and, of course, that works as expected.

I'm deploying Docusaurus within a docker container -- and I think that environment is getting in the way of serving the files appropriately.  Because the files are there in the container at the right location matching the URL.

I'm doing further testing and re-configuring the Dockerfile to try to resolve the issue.

**Reply by boomerbgit** [reply]
The issue has been resolved.  I cannot pinpoint exactly what the resolution was but I updated my Dockerfile to NOT use 'npm run serve' and instead use a self-hosting solution (as recommended in your Deployment pages) using Nginx.
I appreciate you confirming that my design was correct in putting external files into the 'static' folder so they would be picked up by the 'npm run build'.

**Comment by pranabdas** [comment]
@boomerbgit is this workaround good for you? 
```html
<a href="/assets/helm.html" target="_blank">Deploy</a>
```

Seems it only works on production:
```console
npm run build
npm run serve
```
